src/main/java/org/folio/service/CirculationRequestsRetriever.java

```diff
@@ -0,0 +1,76 @@
+package org.folio.service;
+
+import static org.folio.orders.utils.HelperUtils.convertIdsToCqlQuery;
+import static org.folio.rest.RestConstants.MAX_IDS_FOR_GET_RQ_15;
+import static org.folio.service.inventory.InventoryUtils.INVENTORY_LOOKUP_ENDPOINTS;
+import static org.folio.service.inventory.InventoryUtils.REQUESTS;
+import static org.folio.service.inventory.util.RequestFields.COLLECTION_RECORDS;
+import static org.folio.service.inventory.util.RequestFields.COLLECTION_TOTAL;
+import static org.folio.service.inventory.util.RequestFields.ID_KEY;
+import static org.folio.service.inventory.util.RequestFields.ITEM_ID_KEY;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.core.RestClient;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import one.util.streamex.StreamEx;
+
+public class CirculationRequestsRetriever {
+
+  private final RestClient restClient;
+
+  public CirculationRequestsRetriever(RestClient restClient) {
+    this.restClient = restClient;
+  }
+
+  public Future<Integer> getNumberOfRequestsByItemId(String itemId, RequestContext requestContext) {
+    String query = String.format("(itemId==%s and status=\"*\")", itemId);
```

Comment:
regarding status filters, we will ask on refinement do we filter by open ones only.
Here is list of all available statuses:
"Open - Not yet filled",
"Open - Awaiting pickup",
"Open - In transit",
"Open - Awaiting delivery",
"Closed - Filled",
"Closed - Cancelled",
"Closed - Unfilled",
"Closed - Pickup expired"

Reply:
Tested and works for each Open status now